### PR TITLE
Bugfix/windows os error

### PR DESCRIPTION
**Pull request by nekokatt**
Fixes asyncio windows error where "something that is not a socket was shutdown".

### CODE REVIEW

File: hikari/config.py [hunk 1]
```diff
@@ -116,45 +203,182 @@ class HTTPTimeoutSettings:
...
     """
 
+    @acquire_and_connect.validator
+    @request_socket_connect.validator
+    @request_socket_read.validator
+    @total.validator
+    def _(self, attrib: attr.Attribute[typing.Optional[float]], value: typing.Optional[float]) -> None:
+        # This error won't occur until some time in the future where it will be annoying to
+        # try and determine the root cause, so validate it NOW.
+        if value is not None and (not isinstance(value, (float, int)) or value <= 0):  # type: ignore[unreachable]
```
Comment (nekokatt) [inline hikari/config.py:256]: typeshed is nice, it assumes the validators are always getting valid input
that defeats the object of them in the first place surely?

**Review by FasterSpeeding — APPROVED** [review]
lgtm

File: hikari/utilities/ux.py [hunk 2]
```diff
@@ -22,6 +22,10 @@
 """User-experience extensions and utilities."""
 from __future__ import annotations
 
+import time
+
+from hikari.utilities import net
```
Comment (FasterSpeeding) [inline hikari/utilities/ux.py:27]: more miss-placed imports

Comment (nekokatt) [reply hikari/utilities/ux.py:27]: k, fix on other pr.

**Pull request merged**